### PR TITLE
Fixed IDENTITY-6388 IDENTITY-6388

### PR DESCRIPTION
**Pull request by isharak**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java [hunk 1]
```diff
@@ -152,9 +152,11 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                 } else if (errorCode.equals(
                         IdentityCoreConstants.ADMIN_FORCED_USER_PASSWORD_RESET_VIA_OTP_ERROR_CODE)) {
                     String username = request.getParameter(BasicAuthenticatorConstants.USER_NAME);
+                    String tenantDoamin = MultitenantUtils.getTenantDomain(username);
                     String redirectURL = (PASSWORD_RESET_ENDPOINT + queryParams) +
-                            BasicAuthenticatorConstants.USER_NAME + "=" + URLEncoder.encode(username) +
-                            "&confirmation=" + password;
+                            BasicAuthenticatorConstants.USER_NAME_PARAM + URLEncoder.encode(username, BasicAuthenticatorConstants.UTF_8) +
+                            BasicAuthenticatorConstants.TENANT_DOMAIN_PARAM + URLEncoder.encode(tenantDoamin, BasicAuthenticatorConstants.UTF_8) +
+                            BasicAuthenticatorConstants.CONFIRMATION_PARAM + password;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java]: is password value URL safe?

Comment (isharak) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java]: In this case this is not the password. its a used confirmation code.

Comment (mefarazath) [reply components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java]: shouldn't we URL encode it as well?

File: components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java [hunk 2]
```diff
@@ -96,6 +96,7 @@
     private String redirect;
 
     private String dummyUserName = "dummyUserName";
+    private String super_tenant = "carbon.super";
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java]: Shall we use a already existing constant for this?

**Pull request merged**